pkg/kubelet/container_bridge.go

```diff
@@ -17,36 +17,87 @@ limitations under the License.
 package kubelet
 
 import (
-	"bytes"
 	"fmt"
 	"net"
 	"os"
 	"os/exec"
-	"regexp"
+	"syscall"
 
 	"github.com/golang/glog"
+	"github.com/vishvananda/netlink"
+
 	"k8s.io/kubernetes/pkg/util"
 	"k8s.io/kubernetes/pkg/util/iptables"
 	"k8s.io/kubernetes/pkg/util/procfs"
-	"syscall"
 )
 
-var cidrRegexp = regexp.MustCompile(`inet ([0-9a-fA-F.:]*/[0-9]*)`)
+// ensureBridge and ensureIPTablesMasqRule are the only methods that are called from outside this file
+func ensureBridge(brName string, brMtu int, wantCIDR *net.IPNet, promiscuous, babysitDaemons bool) error {
+	br := &netlink.Bridge{
+		LinkAttrs: netlink.LinkAttrs{
+			Name: brName,
+			MTU:  brMtu,
+			// Let kernel use default txqueuelen; leaving it unset
+			// means 0, and a zero-length TX queue messes up FIFO
+			// traffic shapers which use TX queue length as the
+			// default packet limit
+			TxQLen: -1,
+		},
+	}
 
-func createCBR0(wantCIDR *net.IPNet, babysitDaemons bool) error {
-	// recreate cbr0 with wantCIDR
-	if err := exec.Command("brctl", "addbr", "cbr0").Run(); err != nil {
-		glog.Error(err)
+	exists, err := bridgeExists(br.Name)
+	if err != nil {
 		return err
 	}
-	if err := exec.Command("ip", "addr", "add", wantCIDR.String(), "dev", "cbr0").Run(); err != nil {
-		glog.Error(err)
-		return err
+	if !exists {
+		glog.V(2).Infof("%s doesn't exist, attempting to create it with range: %s", br.Name, wantCIDR)
+		return createBridge(br, wantCIDR, babysitDaemons)
 	}
-	if err := exec.Command("ip", "link", "set", "dev", "cbr0", "mtu", "1460", "up").Run(); err != nil {
-		glog.Error(err)
-		return err
+	if !bridgeCidrCorrect(br, wantCIDR) {
+		glog.V(2).Infof("Attempting to recreate %s with address range: %s", br.Name, wantCIDR)
+
+		if err := netlink.LinkSetDown(br); err != nil {
```

Comment:
Don't need to do this with netlink.  Only the ioctl-based interface (which brctl uses) requires setting hte bridge down to delete.